tensorly/decomposition/_tucker.py

```diff
@@ -460,7 +464,7 @@ def non_negative_tucker_hals(tensor, rank, n_iter_max=100, init="svd", svd='nump
     """
     rank = validate_tucker_rank(tl.shape(tensor), rank=rank)
     n_modes = tl.ndim(tensor)
-    if sparsity_coefficients is None or len(sparsity_coefficients) != n_modes:
+    if sparsity_coefficients is None or isinstance(sparsity_coefficients, float):
```

Comment:
```suggestion
    if sparsity_coefficients is None or not isinstance(sparsity_coefficients, Iterable):
```

This requires to import Iterable:
```
from collections.abc import Iterable
```

Reply:
Thanks, I changed this line according to your suggestion.